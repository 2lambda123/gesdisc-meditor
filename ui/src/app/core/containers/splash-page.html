<div fxFill fxLayout="row">
  <div class="sec2" fxFlex></div>
  <div fxFlex fxLayout="column">
      <div class="sec3" fxFlex="calc(30% - 60px)"></div>
      <med-content-type-splash fxLayout="row wrap" fxLayoutAlign="center center"></med-content-type-splash>
      <div class="sec3" fxFlex="calc(30% + 60px)"></div>
  </div>
  <div class="sec2" fxFlex></div>
</div>