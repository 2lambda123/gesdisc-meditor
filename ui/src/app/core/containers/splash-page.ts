import { Component, ChangeDetectionStrategy } from '@angular/core';

@Component({
	selector: 'med-splash-page',
	changeDetection: ChangeDetectionStrategy.OnPush,
	template: `
		<div>it works!</div>
	`
	// TODO: Create component and add it to template
	// <med-content-type></med-content-type>

	,
	styles: [
		`
		:host {
			text-align: center;
		}
	`,
	],
})
export class SplashPageComponent {}
