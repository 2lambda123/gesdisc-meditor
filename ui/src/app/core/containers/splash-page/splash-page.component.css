.content-types-card {
	margin-top: 5px;
}

.admin-types-card {
	margin-top: 20px;
}
