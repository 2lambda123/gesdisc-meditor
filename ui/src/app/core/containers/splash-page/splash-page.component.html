<mat-card *ngFor="let category of categories$ | async">
	<mat-card-title  style="text-align: center; color:gray;">
		{{ category }}
	</mat-card-title>
	<mat-card-content fxLayout="row wrap" fxLayoutAlign="center center">
		<ng-container *ngFor="let model of models$ | async">
			<med-model-button
				style="margin:10px"
				*ngIf="model.category == category"
				[model]="model"
				(goToSearch)="goToSearchPage($event)">
			</med-model-button>
		</ng-container>
	</mat-card-content>
</mat-card>

