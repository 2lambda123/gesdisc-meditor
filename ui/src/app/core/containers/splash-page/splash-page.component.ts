import { Component } from '@angular/core';
import { ModelCatalogEntry } from '../../../service/model/modelCatalogEntry';
import { Observable } from 'rxjs/Observable';
import { Router } from '@angular/router';

import { Store, select } from '@ngrx/store';

import * as fromModel from '../../../reducers';


@Component({
	selector: 'med-splash-page',
	templateUrl: './splash-page.component.html',
	styleUrls: ['./splash-page.component.css']
})
export class SplashPageComponent {

	models$: Observable<ModelCatalogEntry[]>;
	adminModels$: Observable<ModelCatalogEntry[]>;

	constructor(
		private store: Store<fromModel.State>,
		private router: Router
	) {
		this.models$ = store.pipe(select(fromModel.getNonAdminModels));
		this.adminModels$ = store.pipe(select(fromModel.getAdminModels));
	}

	goToSearchPage(event: any) {
		this.router.navigate(['/search'], {queryParams: { model: event.name }});
	}

}
