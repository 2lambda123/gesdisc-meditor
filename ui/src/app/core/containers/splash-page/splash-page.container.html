<div fxFill fxLayout="row">
  <div
    fxFlex.gt-md="calc((100%-600px)/2)"
    fxFlex="calc((100%/3)-1%)"
    fxFlex.xs="calc((100%-200px)/2)">
  </div>
  <div 
    fxFlex.gt-md="600px"
    fxFlex="calc((100%/3)+2%)" 
    fxFlex.xs="200px" 
    fxLayout="column">
      <div
        fxFlex.gt-md="calc(30%-64px)" 
        fxFlex.md="calc(15%-64px)"
        fxFlex.sm="calc((15%/2)-64px)"
        fxFlex.xs="calc((15%/4)-64px)">
      </div>
      <med-splash-box 
        fxFlex="calc(40%)"
        fxLayout="column" 
        fxLayoutAlign.lg="start center">
      </med-splash-box>
      <div 
        fxFlex="calc(30%+64px)">
      </div>
  </div>
  <div
    fxFlex.gt-md="calc((100%-600px)/2)"
    fxFlex="calc((100%/3)-1%)"
    fxFlex.xs="calc((100%-200px)/2)">
  </div>
</div>