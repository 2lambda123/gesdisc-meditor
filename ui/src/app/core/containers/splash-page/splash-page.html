<div fxFill fxLayout="row">
  <div
    fxFlex.gt-xs="calc((100%/3)+1%)"
    fxFlex.xs="calc((100%-200px)/2)">
  </div>
  <div 
    fxFlex.gt-xs="calc((100%/3)-2%)" 
    fxFlex.xs="200px" 
    fxLayout="column">
      <div 
        fxFlex.lg="calc(30%-64px)" 
        fxFlex.lt-lg="calc(15%-64px)">
      </div>
      <med-splash-box 
        fxFlex="calc(40%+64px)"
        fxLayout="column" 
        fxLayoutAlign.lg="start center">
      </med-splash-box>
      <div 
        fxFlex="30%">
      </div>
  </div>
  <div
    fxFlex.gt-xs="calc((100%/3)+1%)"
    fxFlex.xs="calc((100%-200px)/2)">
  </div>
</div>