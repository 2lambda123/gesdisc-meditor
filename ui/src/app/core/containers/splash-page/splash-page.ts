import { Component, ChangeDetectionStrategy } from '@angular/core';

@Component({
	selector: 'med-splash-page',
	changeDetection: ChangeDetectionStrategy.OnPush,
	// template: `
	// 	<div>
	// 		<button mat-fab><mat-icon>warning</mat-icon></button>
	// 		<button mat-fab><mat-icon>description</mat-icon></button>
	// 		<button mat-fab><mat-icon>build</mat-icon></button>
	// 		<button mat-fab><mat-icon>image</mat-icon></button>
	// 		<button mat-fab><mat-icon>info</mat-icon></button>
	// 		<button mat-fab><mat-icon>help</mat-icon></button>
	// 	</div>

	// 	<div>
	// 		<button mat-fab><fa size="2x" name="exclamation-triangle"></fa></button>
	// 		<button mat-fab><fa size="2x" name="newspaper-o"></fa></button>
	// 		<button mat-fab style="background-color: blue;"><fa size="2x" name="newspaper-o"></fa></button>
	// 		<button mat-fab><fa size="2x" name="info"></fa></button>
	// 		<button mat-fab style="background-color: yellow;"><fa size="2x" name="question"></fa></button>
	// 		<button mat-fab><fa size="2x" name="image"></fa></button>
	// 		<button mat-fab style="background-color: green;"><fa size="2x" name="wrench"></fa></button>
	// 		<button mat-fab><fa size="2x" name="file-text"></fa></button>
	// 	</div>
	// `
	
	templateUrl: './splash-page.html'
	,
	styles: [
		`
		:host {
			text-align: center;
		}
	`,
	],
})
export class SplashPageComponent {}
