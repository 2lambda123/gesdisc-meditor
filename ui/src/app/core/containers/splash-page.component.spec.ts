import { ComponentFixture, TestBed } from '@angular/core/testing';
import { combineReducers, Store, StoreModule } from '@ngrx/store';
import { SplashPageComponent } from './splash-page.component';
import { RouterTestingModule } from '@angular/router/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatCardModule, MatButtonModule } from '@angular/material';
import { ModelButtonComponent } from '../components/model-button/model-button.component';
import * as fromModel from '../../reducers';

describe('Splash Page', () => {
  let fixture: ComponentFixture<SplashPageComponent>;
  let store: Store<fromModel.State>;
  let instance: SplashPageComponent;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        NoopAnimationsModule,
        StoreModule.forRoot({
          models: combineReducers(fromModel.reducers),
        }),
        MatCardModule,
        MatButtonModule,
        RouterTestingModule
      ],
      declarations: [
        SplashPageComponent,
        ModelButtonComponent
      ],
    });

    fixture = TestBed.createComponent(SplashPageComponent);
    instance = fixture.componentInstance;
    store = TestBed.get(Store);

    spyOn(store, 'dispatch').and.callThrough();
  });

  it('should compile', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
