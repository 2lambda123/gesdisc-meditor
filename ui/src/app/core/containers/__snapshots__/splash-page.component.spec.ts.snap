// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page should compile 1`] = `
<med-splash-page
  models$={[Function Store]}
  router={[Function Router]}
  store={[Function Store]}
>
  <div
    fxfill=""
    fxlayout="row"
  />
</med-splash-page>
`;
